modules/tribes/tests/client/components/TribesPage.component.test.js

```diff
@@ -0,0 +1,177 @@
+/* tests to write
+
+when clicking join and not logged in, redirect to tribe page
+when clicking join and logged in, send a request to join to API
+when clicking join and logged in, update the number of tribe members
+when clicking leave, send a request to leave to API
+when clicking leave and logged in, update the number of tribe members
+
+
+*/
+
+import React from 'react';
+import * as rtl from '@testing-library/react';
+import { render, fireEvent, wait } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import '@/config/client/i18n';
+
+import TribesPage from '@/modules/tribes/client/components/TribesPage.component';
+import * as api from '@/modules/tribes/client/api/tribes.api';
+
+jest.mock('@/modules/tribes/client/api/tribes.api');
+
+const dummyTribes = [{
+  _id: 'aaaa',
+  id: 'aaaa',
+  count: 20000,
+  slug: 'hitchhikers',
+  label: 'Hitchhikers',
+}, {
+  _id: 'bbbb',
+  id: 'bbbb',
+  count: 5000,
+  slug: 'snails',
+  label: 'Snails',
+}, {
+  _id: 'cccc',
+  id: 'cccc',
+  count: 1500,
+  slug: 'teddybears',
+  label: 'Teddy Bears',
+}];
+
+const dummyUser = {
+  _id: 'user0',
+  id: 'user0',
+  memberIds: ['cccc', 'aaaa'],
+};
+
+api.read.mockImplementation(async () => dummyTribes);
+api.join.mockImplementation(async id => ({ tribe: dummyTribes.find(tribe => tribe._id === id) }));
+api.leave.mockImplementation(async id => ({ tribe: dummyTribes.find(tribe => tribe._id === id) }));
```

Comment:
I have tended to only mock implementations at the exact scope they are required to try and make it clear what will expect it. So inside the test method if it is unique, or in a `beforeEach` in the closest `describe` block.

I am not set on that approach though, so more a comment for consideration.

Reply:
I'll leave this as it is for now. It'll make a lot of sense to move to `beforeEach` when we create the fake data `beforeEach` test. These `mockImplementation`s seem to be bound to those dummyTribes. Does it make sense?